src/main/java/core/basesyntax/SalaryInfo.java

```diff
@@ -1,7 +1,41 @@
 package core.basesyntax;
 
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+
 public class SalaryInfo {
     public String getSalaryInfo(String[] names, String[] data, String dateFrom, String dateTo) {
-        return null;
+        StringBuilder salaryInfo =
+                new StringBuilder("Report for period " + dateFrom + " - "
+                        + dateTo + System.lineSeparator());
+        DateTimeFormatter dateTimeFormatter = DateTimeFormatter.ofPattern("dd.MM.yyyy");
+        LocalDate dateF = LocalDate.parse(dateFrom, dateTimeFormatter);
+        LocalDate dateT = LocalDate.parse(dateTo, dateTimeFormatter);
+        for (String name : names) {
+            int sumOfSalary = 0;
+            for (String datum : data) {
+                String[] parseDatum = datum.split(" ");
+                int salary = 0;
+                for (String datum1 : data) {
```

Comment:
[checklist](https://mate-academy.github.io/jv-program-common-mistakes/java-core/exceptions/salary-info#dont-create-redundant-loops-two-is-enough)